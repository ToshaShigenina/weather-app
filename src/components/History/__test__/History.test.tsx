import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { mountTest, localStorageMock } from '../../../tests';

import { History, type HistoryProps, HISTORY_KEY } from '../';

const testId = 'history';

describe('History', () => {
	beforeAll(() => {
		Object.defineProperty(window, 'localStorage', {
			value: localStorageMock
		});
	});

	mountTest<HistoryProps>(History);

	it('рендерится пустой компонент, в LS устанавливается значение', () => {
		const { baseElement } = render(<History data-testid={ testId } />);
		expect(screen.queryByTestId(testId)).toEqual(null);
		expect(window.localStorage.getItem(HISTORY_KEY)).toEqual('[]');
		expect(baseElement).toMatchSnapshot();
	});
	it('история загружается из LS', () => {
		const historyItems = ['Item 1', 'Item 2', 'Item 3']
		window.localStorage.setItem(HISTORY_KEY, JSON.stringify(historyItems));
		render(<History data-testid={ testId } />);
		expect(screen.getByTestId(testId)).toMatchSnapshot();
	});
})
