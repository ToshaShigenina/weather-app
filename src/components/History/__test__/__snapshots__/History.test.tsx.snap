// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History история загружается из LS 1`] = `
<ul
  class="history"
  data-testid="history"
>
  <li
    class="history__item"
  >
    Item 1
  </li>
  <li
    class="history__item"
  >
    Item 2
  </li>
  <li
    class="history__item"
  >
    Item 3
  </li>
</ul>
`;

exports[`History рендерится пустой компонент, в LS устанавливается значение 1`] = `
<body>
  <div />
</body>
`;
